code/game/objects/items/weapons/medigun.dm

```diff
@@ -0,0 +1,386 @@
+					if(heartcheck)
+						heartcheck.take_damage(rand(5,30))
+
+
+/obj/item/medigun/emag_act(mob/user as mob)
+	if(!emagged)
+		to_chat(user, "<span class='warning'>You swipe the cryptographic sequencer through the circuits.</span>")
+		emagged = 1
+		medigunpack.emagged = 1
+		medigunpack.update_icon()
+		icon_state = "medigunred"
+		item_state = "medigunred"
+		user.update_inv_back()
+		user.update_inv_hands()
+
+/obj/item/medigun/proc/heal(var/mob/living/M)
+	var/healing = MEDIHEAL
+	if(emagged)
+		healing = MEDIHARM
+
+	if(istype(M,/mob/living/carbon/human))
+		var/mob/living/carbon/human/H = M
+		var/datum/organ/internal/heart/heartcheck = H.internal_organs_by_name["heart"]
+		if(heartcheck.modifiers & OMODIFER_UBER)
+			healing = MEDIUBER
+
+	M.adjustOxyLoss(-2*healing)
+	M.heal_organ_damage(healing*2, 0)
+	M.heal_organ_damage(0, healing*2)
+	M.adjustToxLoss(-2*healing)
+	M.adjustCloneLoss(-2*healing)
+	M.adjustBrainLoss(-2*healing)
+	M.adjustBruteLoss(-2*healing)
+	M.adjustFireLoss(-2*healing)
+
+	if(healing > 0)
+		if(M.reagents)
+			var/list/reagents_to_check = list(
+				"toxin",
+				"stoxin",
+				"plasma",
+				"sacid",
+				"pacid",
+				"cyanide",
+				"amatoxin",
+				"chloralhydrate",
+				"carpotoxin",
+				"mindbreaker",
+				)
+			for(var/reagent in reagents_to_check)
+				if(M.reagents.has_reagent(reagent))
+					M.reagents.remove_reagent(reagent, healing)
+
+		if(istype(M,/mob/living/simple_animal))
+			var/mob/living/simple_animal/SA = M
+			if(SA.health > 0)
+				SA.health = min(SA.maxHealth,SA.health+2)
+
+		if(istype(M,/mob/living/carbon/human))
+			var/mob/living/carbon/human/H = M
+			for(var/datum/organ/external/temp in H.organs)
+				if(temp.status & ORGAN_BLEEDING)
+					temp.clamp()
+
+		if(M.losebreath >= 10)
+			M.losebreath = max(10, M.losebreath - 5)
+
+		if(M.bodytemperature > 310)
+			M.bodytemperature = max(310, M.bodytemperature - (40 * TEMPERATURE_DAMAGE_COEFFICIENT))
+		else if(M.bodytemperature < 311)
+			M.bodytemperature = min(310, M.bodytemperature + (40 * TEMPERATURE_DAMAGE_COEFFICIENT))
+
+	M.UpdateDamageIcon()
+	var/tracker_effect = "heal"
+	if(emagged)
+		tracker_effect += "red"
+	make_tracker_effects(get_turf(src), M, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+	spawn(2)
+		make_tracker_effects(get_turf(src), M, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+	spawn(4)
+		make_tracker_effects(get_turf(src), M, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+	spawn(6)
+		make_tracker_effects(get_turf(src), M, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+	spawn(8)
+		make_tracker_effects(get_turf(src), M, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+	if(healing == MEDIUBER)
+		if((ubercharge == (MAX_UBERCHARGE-1)) && istype(loc,/mob))
+			to_chat(loc,"<span class='info'>CHARGE READY</span>")
+			playsound(loc, 'sound/weapons/medigun_charged.ogg', 50, 1)
+		ubercharge = min(ubercharge+1,MAX_UBERCHARGE)
+
+/obj/item/medigun/dropped(mob/user)
+	spawn(1)//to prevent a few exploits
```

Comment:
you know what? so did I. So I'll remove but feel free to re-add it if it breaks something.